app/models/piece.rb

```diff
@@ -1,2 +1,8 @@
 class Piece < ApplicationRecord
+  belongs_to :game
+  self.inheritance_column :types
+
+  def self.types
```

Comment:
I'm wondering if this method is necessary. It seems that the way STI works in rails is that it automatically looks for subclasses with the name in the type field. By the same token, the above "self.inheritance_column" shouldn't be necessary (please correct me if I'm wrong). Either way, not a big deal...